src/commands/Quote.ts

```diff
@@ -0,0 +1,207 @@
+import { Client, Message} from "discord.js";
+// import { connect } from "http2";
+import { ICommand } from "./Command";
+const fs = require('fs');
+/*
+  Extension of the base Command interface for the Impersonate command.
+  The impersonate command only needs the message to echo.
+*/
+
+//The types of commands are as followed
+//:quote 
+//:quote find #2
+//:quote find "yolo"
+//:quote find
+//:quote random
+//:quote delete
+
+//TheLegend27 = 0
+//Edison = 1
+//Satvik = 2
+//NKR = 3
+//AweKingSome = 4
+
+export interface IQuoteCommand extends ICommand {
+  generationType: string;
+  findType: string;
+  findTypeSize: number;
+}
+  
+export class QuoteCommand implements IQuoteCommand {
+    discordMessage?: Message;
+    generationType: string;
+    findType: string;
+    findTypeSize: number;
+    constructor(generationType: string, findType: string, findTypeSize: number){
+      this.generationType = generationType;
+      this.findType = findType;
+      this.findTypeSize = findTypeSize;
+    }
+    execute(_client?: Client) {
+      if(this.discordMessage?.reference?.messageID != null){
+          this.discordMessage?.channel.messages.fetch(this.discordMessage?.reference?.messageID!)
+          .then(message => {
+            //Helpfull PreDebugging Tool
+            // console.log("findType")
+            // console.log(this.findType)
+            // console.log("findTypeSize")
+            // console.log(this.findTypeSize)
+            // console.log("generationType")
+            // console.log(this.generationType)
+            var value = -1
+            if (message.author.username == "TheLegend27"){
+              value = 0;
+            } else if (message.author.username == "Edison"){
+              value = 1;
+            } else if (message.author.username == "Satvik"){
+              value = 2;
+            } else if (message.author.username == "NKR"){
+              value = 3
+            } else if (message.author.username == "AweKingSome"){
+              value = 4
+            }
```

Comment:
Do not hardcode usernames. Rename "value" to something more meaningful. 
I suggest to have "value" be the discord user ID of the message author.

Reply:
Done